data/org.freedesktop.portal.WebExtensions.xml

```diff
@@ -0,0 +1,129 @@
+    -->
+    <method name="CreateSession">
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="session_handle" direction="out"/>
+    </method>
+    <!--
+        Start:
+        @session_handle: Object path for the #org.freedesktop.portal.Session object
+        @name: name of the native messaging server
+        @extension_or_origin: extension ID or origin URI identifying the extension
+        @json_manifest: the JSON manifest for the native messaging server
+
+        Return the JSON manifest of the native messaging server that
+        would Start would invoke.
+    -->
+    <method name="GetManifest">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="s" name="name" direction="in"/>
+      <arg type="s" name="extension_or_origin" direction="in"/>
+      <arg type="s" name="json_manifest" direction="out"/>
+    </method>
+    <!--
+        Start:
+        @session_handle: Object path for the #org.freedesktop.portal.Session object
+        @name: name of the native messaging server
+        @extension_or_origin: extension ID or origin URI identifying the extension
+        @options: Vardict with optional further information
+        @handle: Object path for the #org.freedesktop.portal.Request object representing this call
+
+        Start the named native messaging server. The caller must
+        indicate the requesting web extension (either by extension ID
+        for Firefox, or origin URI for Chrome), which will be matched
+        against the server's access control list.
+
+        If the server can't be started, or invalid data is provided,
+        the session will be closed.
+
+        Supported keys in the @options vardict include:
+        <variablelist>
+          <varlistentry>
+            <term>handle_token s</term>
+            <listitem><para>
+              A string that will be used as the last element of the @handle. Must be a valid
+              object path element. See the #org.freedesktop.portal.Request documentation for
+              more information about the @handle.
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <method name="Start">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="s" name="name" direction="in"/>
+      <arg type="s" name="extension_or_origin" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="handle" direction="out"/>
+    </method>
+    <!--
+        GetPipes:
+        @session_handle: Object path for the #org.freedesktop.portal.Session object
+        @options: Vardict with optional further information
+        @stdin: File descriptor representing the server's stdin.
+        @stdout: File descriptor representing the server's stdout.
+        @stderr: File descriptor representing the server's stderr.
+
+        Retrieve file descriptors for the native messaging server
+        identified by the session.
```

Comment:
Perhaps mention this must be called after the `Start` request got a "ok" response signal?